tools/filter_cells_and_genes/filter_cells_and_genes.xml

```diff
@@ -0,0 +1,270 @@
+    </test>
+
+    <test>
+         <param name="file" value="counts.tab" ftype="tabular"/>
+         <param name="file_sep" value="tab"/>
+         <param name="file_header" value="TRUE"/>
+         <param name="is_mito" value="FALSE"/>
+         <param name="package" value="Seurat"/>
+         <param name="min_genes" value="10"/>
+         <param name="min_cells" value="3"/>
+         <output name="output_plot" file="counts_Seurat_filter_plots.pdf" ftype="pdf"/>
+         <output name="output_matrix" file="counts_Seurat_filtered.tab" ftype="tabular"/>
+    </test>
+    <test>
+         <param name="file" value="counts_with_mito.tab" ftype="tabular"/>
+         <param name="file_sep" value="tab"/>
+         <param name="file_header" value="TRUE"/>
+         <param name="is_mito" value="TRUE"/>
+         <param name="header_mito" value="MT-"/>
+         <param name="package" value="scater"/>
+         <param name="gene_filter" value="low.abundances"/>
+         <param name="min_cells" value="1"/>
+         <output name="output_plot" file="counts_with_mito_scater_low.abundances_filter_plots.pdf" ftype="pdf"/>
+         <output name="output_matrix" file="counts_with_mito_scater_low.abundances_filtered.tab" ftype="tabular"/>
+    </test>
+    <test>
+         <param name="file" value="counts_with_mito.tab" ftype="tabular"/>
+         <param name="file_sep" value="tab"/>
+         <param name="file_header" value="TRUE"/>
+         <param name="is_mito" value="TRUE"/>
+         <param name="header_mito" value="MT-"/>
+         <param name="package" value="scater"/>
+         <param name="gene_filter" value="min.cells"/>
+         <param name="min_cells" value="3"/>
+         <output name="output_plot" file="counts_with_mito_scater_min.cells_filter_plots.pdf" ftype="pdf"/>
+         <output name="output_matrix" file="counts_with_mito_scater_min.cells_filtered.tab" ftype="tabular"/>
+    </test>
+    <test>
+         <param name="file" value="counts_with_mito.tab" ftype="tabular"/>
+         <param name="file_sep" value="tab"/>
+         <param name="file_header" value="TRUE"/>
+         <param name="is_mito" value="TRUE"/>
+         <param name="header_mito" value="MT-"/>
+         <param name="package" value="Seurat"/>
+         <param name="min_genes" value="10"/>
+         <param name="min_cells" value="3"/>
+         <output name="output_plot" file="counts_with_mito_Seurat_filter_plots.pdf" ftype="pdf"/>
+         <output name="output_matrix" file="counts_with_mito_Seurat_filtered.tab" ftype="tabular"/>
+    </test> 
+   </tests>
+   <help>
+
+**What it does**
+
+Filter_cells_and_genes takes a tabular table of raw read counts, each column corresponding to a cell and each row corresponding to a
+gene, and returns (1) a PDF with quality controls plots (like the distribution of library sizes or expressed genes by cell, frequency of 
+expression or the most expressed genes) and (2) a filtered table of counts.
+
```

Comment:
(1) a PDF with quality controls plots (see explanations below) and (2) a filtered table of counts.

Reply:
Ping @bellenger-l